crates/uniswapx-rs/src/order.rs

```diff
@@ -239,7 +288,73 @@ impl PriorityOutput {
             recipient: self.recipient.to_string(),
         }
     }
+}
 
+impl V3DutchOrder {
+    pub fn decode_inner(order_hex: &[u8], validate: bool) -> Result<Self, Box<dyn Error>> {
+        Ok(V3DutchOrder::decode_single(order_hex, validate)?)
+    }
+
+    pub fn encode_inner(&self) -> Vec<u8> {
+        V3DutchOrder::encode_single(self)
+    }
+
+    pub fn resolve(&self, block_number: u64, timestamp: u64) -> OrderResolution {
+        let timestamp = Uint::from(timestamp);
+
+        if self.info.deadline.lt(&timestamp) {
+            return OrderResolution::Expired;
+        };
+
+        // resolve over the decay curve
+        // TODO: apply cosigner logic
```

Comment:
Pulled this from V2.. @zhongeric I see the cosigner overrides are being used and exclusivity is accounted for.. is there anything else that needs to be done?

Reply:
hmm seems fine to me